src/main/java/com/smartbear/postman/utils/RestServiceCreator.java

```diff
@@ -40,68 +40,66 @@
+        uri = request.getUrl();
+        rawModeData = request.getBody();
 
-        String rawModeData = request.getBody();
+        PostmanRestServiceBuilder builder = new PostmanRestServiceBuilder();
 
-        RestRequest restRequest = addRestRequest(project, request.getMethod(), uri, request.getHeaders());
-        if (restRequest == null) {
+        RestRequest restRequest;
+        try {
+            restRequest = builder.createRestServiceFromPostman(request.getMethod(), request.getHeaders());
+        } catch (Exception e) {
+            logger.error("Error while creating a REST service", e);
             return null;
         }
 
+        String requestName = request.getName();
         if (StringUtils.hasContent(requestName)) {
             restRequest.setName(requestName);
         }
 
-        if (HttpUtils.canHavePayload(restRequest.getMethod()) && StringUtils.hasContent(rawModeData)) {
-            if ("formdata".equals(request.getMode())) {
-                JSON dataJson = new PostmanJsonUtil().parseTrimmedText(rawModeData);
-                if (dataJson instanceof JSONArray) {
-                    JSONArray dataArray = (JSONArray) dataJson;
-                    Arrays.stream(dataArray.toArray()).forEach(d -> {
-                        if (d instanceof JSONObject) {
-                            JSONObject object = (JSONObject) d;
-                            String key = object.getString("key");
-                            String value = object.getString("value");
-
-                            restRequest.getParams().addProperty(key);
-                            restRequest.getParams().setPropertyValue(key, value);
-                        }
-                    });
-                }
-                restRequest.setMediaType(MediaType.MULTIPART_FORM_DATA);
-                restRequest.setPostQueryString(true);
-            } else {
-                restRequest.setRequestContent(rawModeData);
-            }
-        }
+        addRequestBody(request, restRequest);
         return restRequest;
     }
 
-    private RestRequest addRestRequest(WsdlProject project, String method, String uri, List<PostmanCollection.Header> headers) {
-        RestRequest currentRequest = null;
-        PostmanRestServiceBuilder builder = new PostmanRestServiceBuilder();
-        try {
-            currentRequest = builder.createRestServiceFromPostman(project, uri,
-                    RestRequestInterface.HttpMethod.valueOf(method), headers);
-        } catch (Exception e) {
-            logger.error("Error while creating a REST service", e);
+    private void addRequestBody(Request request, RestRequest restRequest) {
+        if (!HttpUtils.canHavePayload(restRequest.getMethod()) || !StringUtils.hasContent(rawModeData)) {
+            return;
+        }
+        if ("formdata".equals(request.getMode())) {
+            JSON dataJson = new PostmanJsonUtil().parseTrimmedText(rawModeData);
+            if (dataJson instanceof JSONArray) {
```

Comment:
Please change to use inline initialization of instanceof variables.
```
            if (dataJson instanceof JSONArray dataArray) {
```
Similarly do it for 2nd usage of instanceof in this method.

Reply:
Currently we are using java 8 in this plugin, we can do it after upgrade to 17